src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -95,20 +132,396 @@ await signInManager.ValidateSecurityStampAsync(refreshTicket.Principal) is not T
+                }
+            }
+            catch (FormatException)
+            {
+                return TypedResults.Unauthorized();
+            }
+
+            if (!result.Succeeded)
+            {
+                return TypedResults.Unauthorized();
+            }
+
+            return TypedResults.Text("Thank you for confirming your email.");
+        })
+        .Add(endpointBuilder =>
+        {
+            var finalPattern = ((RouteEndpointBuilder)endpointBuilder).RoutePattern.RawText;
+            confirmEmailEndpointName = $"{nameof(MapIdentityApi)}-{finalPattern}";
+            endpointBuilder.Metadata.Add(new EndpointNameMetadata(confirmEmailEndpointName));
+            endpointBuilder.Metadata.Add(new RouteNameMetadata(confirmEmailEndpointName));
+        });
+
+        routeGroup.MapPost("/resendConfirmationEmail", async Task<Ok>
+            ([FromBody] ResendEmailRequest resendRequest, [FromServices] IServiceProvider sp) =>
+        {
+            var userManager = sp.GetRequiredService<UserManager<TUser>>();
+            if (await userManager.FindByEmailAsync(resendRequest.Email) is not { } user)
+            {
+                return TypedResults.Ok();
+            }
+
+            await SendConfirmationEmailAsync(user, userManager, resendRequest.Email);
+            return TypedResults.Ok();
+        });
+
+        routeGroup.MapPost("/resetPassword", async Task<Results<Ok, ValidationProblem>>
+            ([FromBody] ResetPasswordRequest resetRequest, [FromServices] IServiceProvider sp) =>
+        {
+            var userManager = sp.GetRequiredService<UserManager<TUser>>();
+
+            if (!string.IsNullOrEmpty(resetRequest.ResetCode) && string.IsNullOrEmpty(resetRequest.NewPassword))
+            {
+                return CreateValidationProblem("MissingNewPassword", "A password reset code was provided without a new password.");
+            }
+
+            var user = await userManager.FindByEmailAsync(resetRequest.Email);
+
+            if (user is null || !(await userManager.IsEmailConfirmedAsync(user)))
+            {
+                // Don't reveal that the user does not exist or is not confirmed, so don't return a 200 if we would have
+                // returned a 400 for an invalid code given a valid user email.
+                if (!string.IsNullOrEmpty(resetRequest.ResetCode))
+                {
+                    return CreateValidationProblem(IdentityResult.Failed(userManager.ErrorDescriber.InvalidToken()));
+                }
+            }
+            else if (string.IsNullOrEmpty(resetRequest.ResetCode))
+            {
+                var code = await userManager.GeneratePasswordResetTokenAsync(user);
+                code = WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(code));
+
+                await emailSender.SendEmailAsync(resetRequest.Email, "Reset your password",
+                    $"Reset your password using the following code: {HtmlEncoder.Default.Encode(code)}");
```

Comment:
I assume we'll want to localize these strings in the future?

Reply:
Probably. It doesn't look like AddDefaultIdentity does in the [password reset razor page](https://github.com/dotnet/aspnetcore/blob/f9828517e57769959d43c89d06ad7f319880258c/src/Identity/UI/src/Areas/Identity/Pages/V5/Account/ForgotPassword.cshtml.cs#L84-L87), but it's worth considering. https://github.com/dotnet/aspnetcore/issues/49664